litex/soc/cores/spiopi.py

```diff
@@ -0,0 +1,1063 @@
+from litex.soc.interconnect.csr_eventmanager import *
+from litex.soc.interconnect import wishbone
+from litex.soc.integration.doc import AutoDoc, ModuleDoc
+from migen.genlib.cdc import MultiReg
+
+class SpiOpi(Module, AutoCSR, AutoDoc):
```

Comment:
This device is Series7 specific right? It would be nice if the device specific part was separated from the generic stuff.

Reply:
Yes, to get it to close timing at 100MHz, you need to do vendor-specific I/O.

To clarify, are you suggesting that the core should not be included here because it is device-specific (which is fine with me), or are you suggesting there should be a "generic" implementation for every vendor-specific part? If it's the latter, the problem is, you won't meet timing, or you'd have to reduce the clock rate on your system so much the core doesn't make sense any more to use...

Reply:
@mithro: i'm fine doing this and adding a banner to specify it's only for 7-Series for now. @bunnie: That's fine having device specific cores/parts, as @mithro suggest, some parts of the core could probably be reused for others devices, but i'm happy doing the work on easing reuse and don't want to bother you with that.

Reply:
@bunnie - This looks really useful and a great addition! I was suggesting that the core could be split into 7-series and generic parts. **Other** people can then create the equivalent of the 7-series bits for the iCE40 / ECP5 / etc.